app/src/main/java/com/google/jetpackcamera/ui/JcaApp.kt

```diff
@@ -34,34 +38,79 @@ import com.google.jetpackcamera.settings.SettingsScreen
 import com.google.jetpackcamera.settings.VersionInfoHolder
 import com.google.jetpackcamera.ui.Routes.PREVIEW_ROUTE
 import com.google.jetpackcamera.ui.Routes.SETTINGS_ROUTE
+import com.google.jetpackcamera.ui.permissions.PermissionEnum
+import com.google.jetpackcamera.ui.permissions.PermissionsScreen
+import com.google.jetpackcamera.ui.permissions.PermissionsViewModel
 
 @OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun JcaApp(
+    openAppSettings: () -> Unit,
     /*TODO(b/306236646): remove after still capture*/
     previewMode: PreviewMode,
     onPreviewViewModel: (PreviewViewModel) -> Unit,
     onRequestWindowColorMode: (Int) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    val permissionState =
-        rememberPermissionState(permission = Manifest.permission.CAMERA)
+    val cameraPermissionState = rememberPermissionState(permission = Manifest.permission.CAMERA)
+    val permissionStates = rememberMultiplePermissionsState(
+        permissions = listOf(
+            Manifest.permission.CAMERA,
+            Manifest.permission.RECORD_AUDIO
+        )
+    )
 
-    if (permissionState.status.isGranted) {
+    // should show rationale means a setting has been viewed and denied
+    if (cameraPermissionState.status.isGranted && (
+            permissionStates.allPermissionsGranted ||
+                permissionStates.shouldShowRationale
+            )
+    ) {
         JetpackCameraNavHost(
             onPreviewViewModel = onPreviewViewModel,
             previewMode = previewMode,
             onRequestWindowColorMode = onRequestWindowColorMode,
             modifier = modifier
         )
     } else {
-        CameraPermission(
+        val permissionsViewModel =
+            viewModel { PermissionsViewModel(getRequestablePermissions(permissionStates)) }
+        // you'll have the option to go through camera and all other optional permissions
+        PermissionsScreen(
             modifier = modifier.fillMaxSize(),
-            cameraPermissionState = permissionState
+            viewModel = permissionsViewModel,
+            openAppSettings = openAppSettings
```

Comment:
It would be preferable to get this into JetpackCameraNavHost so we're more consistent with how we handle our screens.